### DISCUSSION TITLE
TestContainers with Azure Container Registry

### DISCUSSION THREAD

**Discussion by JMAR-TDI-PDT** [testcontainers/testcontainers-dotnet#1123] [Q&A]
Hi,

Is it possible to user Azure Container Registry with testcontainers ? I would like to run units tests on Azure and store in Azure Container Registry images (Sql Server, Rabbitmq for example) and to use like this : 

```
RabbitMqcontainer = new RabbitMqBuilder().WithImage("IMAGE FROM Azure Container Registry")
                                                .WithPortBinding(5672, true)
                                                .WithPortBinding(15672, true)
                                                .WithCommand("rabbitmq-server", "rabbitmq-plugins", "enable", "rabbitmq_management")
                                                .Build();
```

thanks

**Comment by HofmeisterAn** [comment]
If you are running on CI (Azure DevOps or GitHub), you can use the Docker login task to log into the remote registry. The identity token is not yet fully implemented. The pull request https://github.com/testcontainers/testcontainers-dotnet/pull/842 lacks tests. I cannot promise it, but perhaps I can add them by the weekend (I need an ACR first). This is another workaround: https://github.com/testcontainers/testcontainers-dotnet/issues/841#issuecomment-1777449393.
